src/runtime/storage/pouchdb/pouch-db-storage.ts

```diff
@@ -9,7 +9,7 @@
  */
 
 import {assert} from '../../../platform/assert-web.js';
-import {PouchDB, PouchDbDebug, PouchDbMemory} from '../../../platform/pouchdb-web.js';
+import {PouchDB, PouchDbDebug, PouchDbMemory} from '../../../../concrete-storage/pouchdb.js';
```

Comment:
I'm curious, do you have a plan for these files in `src/runtime/pouchdb`? It looks to me like this adds a dep on pouchdb to the runtime code. Or am I misunderstanding?

Reply:
Ultimately, modules that use pouchdb/firebase apis need to depend on the concrete-storage. 

Generally, runtime attaches to storage-providers via `StorageProviderFactory` and never depend on any database-specific storage modules. 

Instead, a particular entry-point (app) can import `storage/<database>/<database>-provider.js` and that will pull in all the dependencies and register the database which `StorageProviderFactory.`

If some runtime api depends on this module (directly or indirectly), then that is a problem.

Reply:
Discussed a bit offline, but to summarise:

"IMO, it would make sense to move the providers up a level so they are sibling with runtime, as they are not part of runtime." -- I agree. Runtime shouldn't have dependencies outside its folder. Instead, other packages should augment the runtime with additional functionality/plugins.